src/object/explosion.cpp

```diff
@@ -171,7 +164,8 @@ Explosion::update(float )
       break;
 
     case E_STATE_EXPLODING:
-      m_color.alpha = std::min(m_fading_timer.get_progress(), 1.f);
+      if (m_light_sprite)
+        m_light_sprite->get_color().alpha = std::min(m_fading_timer.get_progress(), 1.f);
```

Comment:
So if sprites can have their own color then why is `Candle` for example using its own `m_lightcolor`?